src/truncated/normal.jl

```diff
@@ -143,9 +143,18 @@ function rand(rng::AbstractRNG, d::Truncated{Normal{T},Continuous}) where T <: R
     μ = mean(d0)
     σ = std(d0)
     if isfinite(μ)
-        a = (d.lower - μ) / σ
-        b = (d.upper - μ) / σ
-        z = randnt(rng, a, b, d.tp)
+        a, b = extrema(d)
+        if iszero(σ)
+            if a <= μ <= b
+                z = 0.0
+            else
+                return NaN
```

Comment:
I think we would want to catch this case earlier in `truncated`: A distribution with zero mass is by definition not a distribution. See https://github.com/JuliaStats/Distributions.jl/issues/843.

Reply:
Maybe let's throw an error here? I don't think returning `NaN` is particularly useful.

Reply:
I've opened https://github.com/JuliaStats/Distributions.jl/pull/1723 as a means of catching this case earlier, but I'd also be fine throwing an error here.